doc/duckarrays.rst

```diff
@@ -0,0 +1,53 @@
+.. currentmodule:: xarray
+
+Working with numpy-like arrays
+==============================
+
+.. warning::
+
+   This is a experimental feature, please report any bugs you might find.
+
+Numpy-like arrays (:term:`duck array`) extend the :py:class:`numpy.ndarray` with
+additional features, like propagating physical units or a different layout in memory.
+
+:py:class:`DataArray` and :py:class:`Dataset` objects can wrap these duck arrays, as
+long as they satisfy certain conditions (see :ref:`internals.duck_arrays`).
```

Comment:
would it make sense to point to the explicitly tested duck arrays here (`pint`, `sparse`)? We could also add a user-maintained list of duck array libraries, just like the current "related projects" list.

I did think about adding usage examples, but maybe it's better to leave that to the extension packages?

Reply:
I think a list would be nice.